app/controllers/public_recipes_controller.rb

```diff
@@ -0,0 +1,13 @@
+class PublicRecipesController < ApplicationController
+  before_action :authenticate_user!
+  def index
+    @recipes = Recipe.where(public: true).order(created_at: :desc)
```

Comment:
- [OPTIONAL] Please fix the N + 1 queries issue presented by the bullet by adding the `.includes(:user)` to the query. It's a project requirement.